src/lib.rs

```diff
@@ -194,10 +194,9 @@
 //! for Async Rust. We have a long journey ahead of us. But we're excited you're here with us!
 
 #![cfg_attr(feature = "docs", feature(doc_cfg))]
-// #![warn(missing_docs)]
-#![warn(missing_debug_implementations, rust_2018_idioms)]
-#![doc(test(attr(deny(rust_2018_idioms, warnings))))]
-#![doc(test(attr(allow(unused_extern_crates, unused_variables))))]
```

Comment:
These two lines are needed to [catch the doc test warnings](https://doc.rust-lang.org/nightly/rustdoc/the-doc-attribute.html#testattr). (Normal crate-level lints cannot catch doc test warnings.) So you probably don't need to remove them.